File: Basics/Linked-List/linked-list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def insert(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node

    def delete(self, key):
        current = self.head

        if current is not None and current.data == key:
            self.head = current.next
            return

        prev = None
        while current is not None and current.data != key:
            prev = current
            current = current.next

        if current is None:
            return

        prev.next = current.next

# Example usage:
# ll = LinkedList()
# ll.insert(3)
# ll.insert(7)
# ll.delete(3)

File: Basics/Linked-List/linked-list.rs
struct Node {
    data: i32,
    next: Option<Box<Node>>,
}

struct LinkedList {
    head: Option<Box<Node>>,
}

impl LinkedList {
    fn new() -> Self {
        LinkedList { head: None }
    }

    fn insert(&mut self, data: i32) {
        let new_node = Node {
            data,
            next: self.head.take(),
        };
        self.head = Some(Box::new(new_node));
    }

    fn delete(&mut self, key: i32) {
        let mut current = &mut self.head;

        while let Some(node) = current {
            if node.data == key {
                let next = node.next.take();
                *current = next;
                return;
            }
            current = &mut node.next;
        }
    }
}

// Example usage:
// let mut ll = LinkedList::new();
// ll.insert(3);
// ll.insert(7);
// ll.delete(3);
